.github/actions/checkDeployBlockers/checkDeployBlockers.js

```diff
@@ -0,0 +1,46 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+const {GITHUB_OWNER, EXPENSIFY_CASH_REPO} = require('../../libs/GithubUtils');
+
+const run = function () {
+    const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+    const issueNumber = Number(core.getInput('ISSUE_NUMBER', {required: true}));
+    let hasDeployBlockers = false;
+
+    return octokit.rest.issues.get({
+        owner: GITHUB_OWNER,
+        repo: EXPENSIFY_CASH_REPO,
+        issue_number: issueNumber,
+    })
+        .then((issue) => {
+            const body = issue.data.body || '';
+            const pattern = /-\s\[\s]/g;
+            const matches = pattern.exec(body);
+            hasDeployBlockers = matches !== null;
+
+            return issue.data;
+        })
+        .then(() => {
```

Comment:
I sort of think that nested `then` is just a bit of an anti-pattern, and generally speaking we should only enter a new `then` block when we reach something async. So I would structure it like:

```javascript
.then(({data}) => {
    // Look for matches
    
    // If any are found, log them, set the output, and return
    
    // Now it's time to look for :shipit:, so return that api request
    return octokit.issues.listComments(...);
})
.then(({data}) => {
    // data is now the response of `listComments`
});
```

Reply:
Very nice! I'm still wrapping my head around how this async syntax works.